### PR TITLE
Create new color sensor code

### PR DESCRIPTION
**Pull request by selym3**
* `WColor` enum
* Re implemented commands
* Cleaned commands

### CODE REVIEW

File: src/main/java/com/stuypulse/robot/subsystems/ColorSensor.java [hunk 1]
```diff
@@ -9,55 +9,140 @@
...
+        YELLOW,
+        CYAN,
+        GREEN,
+        RED,
+        
+        // To avoid null pointer exceptions when returning error states
+        NONE;
+
+        // Each color will store information about its position
+        private WColor next, rotated;
```
Comment (Sam-Belliveau) [inline src/main/java/com/stuypulse/robot/subsystems/ColorSensor.java]: remove rotated

File: src/main/java/com/stuypulse/robot/subsystems/ColorSensor.java [hunk 2]
```diff
@@ -9,55 +9,140 @@
...
+        // Use a static block to setup circular references
+        static {
+            /*
+            Control Panel Reference: 
+            https://firstfrc.blob.core.windows.net/frc2021/Manual/Sections/2021FRCGameManualSection03.pdf
+            */
+
+            // Setup Yellow
+            YELLOW.next = CYAN; 
+            YELLOW.rotated = GREEN;
```
Comment (Sam-Belliveau) [inline src/main/java/com/stuypulse/robot/subsystems/ColorSensor.java]: don't define rotated here

File: src/main/java/com/stuypulse/robot/subsystems/ColorSensor.java [hunk 3]
```diff
@@ -9,55 +9,140 @@
...
+            NONE.rotated = NONE;
+        }
+
+        // Getters 
+        public WColor getNextColor() {
+            return this.next;
+        }
+
+        public WColor getRotatedColor() {
+            return this.rotated;
```
Comment (Sam-Belliveau) [inline src/main/java/com/stuypulse/robot/subsystems/ColorSensor.java]: change this to `this.next.next`

**Review by Sam-Belliveau — CHANGES_REQUESTED** [review]

**Comment by Sam-Belliveau** [comment]
I like this alot

File: src/main/java/com/stuypulse/robot/subsystems/ColorSensor.java (hunk 3 above)
Comment (selym3) [reply src/main/java/com/stuypulse/robot/subsystems/ColorSensor.java]: Thanks for pointing this out, I changed it

File: src/main/java/com/stuypulse/robot/commands/WoofSpinToFMSColorCommand.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    public void execute() {
+        final WColor goal = woof.getTargetColor(); // <-- reads from the FMS
+        final WColor curr = woof.getCurrentColor();
+
+        // If any color is not reporting, skip 
+        if (goal == WColor.NONE || curr == WColor.NONE)
+            return;
+        
+        // If the goal is behind us (1 away), spin backwards
+        if (curr.getNextColor() == goal)
```
Comment (Sam-Belliveau) [inline src/main/java/com/stuypulse/robot/commands/WoofSpinToFMSColorCommand.java]: this should be `goal.getNextColor() == curr`

**Review by Sam-Belliveau — CHANGES_REQUESTED** [review]

File: src/main/java/com/stuypulse/robot/commands/WoofSpinToFMSColorCommand.java (hunk 4 above)
Comment (selym3) [reply src/main/java/com/stuypulse/robot/commands/WoofSpinToFMSColorCommand.java]: I believe this depends on whichever way the motor spins the control panel so it's up to testing.

**Pull request merged**